llvm/lib/CodeGen/MachineInstr.cpp

```diff
@@ -555,6 +555,16 @@ uint32_t MachineInstr::copyFlagsFromInstruction(const Instruction &I) {
       MIFlags |= MachineInstr::MIFlag::NoUWrap;
   }
 
+  // Copy the nonneg flag.
+  if (const PossiblyNonNegInst *PNI = dyn_cast<PossiblyNonNegInst>(&I))
+    if (PNI->hasNonNeg())
+      MIFlags |= MachineInstr::MIFlag::NonNeg;
+
+  // Copy the disjoint flag.
+  if (const PossiblyDisjointInst *PD = dyn_cast<PossiblyDisjointInst>(&I))
```

Comment:
else if? these are mutually exclusive?

Reply:
I followed the surround code style.

Reply:
I would prefer to keep it like this: for documentation and whenever the flags change again.

Reply:
The possible set of flag changes isn't unbounded

Reply:
Done.